### ISSUE TITLE
Probability of outputs not ensure

### ISSUE THREAD

**Issue by zhuyi3625** [alvinwan/neural-backed-decision-trees#15]
it has different outputs when I input the same picture into the model which is trained by myself.However when use the pretrained '.pth' model,the output is right,and the probability tensor is ensure,in contrast,my probability tensor will change in every tests.

I trained model use **python main.py --lr=0.01 --dataset=CIFAR10 --arch=wrn28_10_cifar10 --hierarchy=induced-wrn28_10_cifar10 --pretrained --loss=SoftTreeSupLoss.**

out tensor([[5.5736e-04, 1.7653e-04, 7.0383e-04, 3.4651e-04, 2.5735e-04, 9.9691e-01,
         9.0089e-05, 6.4592e-04, 1.1384e-04, 2.0057e-04]],
       grad_fn=<CopySlices>)

out tensor([[5.5736e-04, 1.7653e-04, 7.0383e-04, 3.4651e-04, 2.5735e-04, 9.9691e-01,
         9.0089e-05, 6.4592e-04, 1.1384e-04, 2.0057e-04]],
       grad_fn=<CopySlices>)

**Comment by alvinwan** [comment]
Hm are those the outputs after training the model twice? If so, double-check the script downloaded/loaded the pretrained WRN28x10 correctly.

Or do you mean those are the outputs after training *1 model and running inference twice? If so, make sure you're using `model.eval()` for deterministic predictions.
